pkg/server/pod_controller.go

```diff
@@ -491,6 +449,29 @@ func (c *PodController) dispatchPodToNode(pod *api.Pod, node *api.Node) {
 	}
 }
 
+func (c *PodController) addCloudRoute(node *api.Node, cidrs []string) error {
+	instanceID := node.Status.InstanceID
+	err := c.cloudClient.ModifySourceDestinationCheck(instanceID, false)
+	if err != nil {
+		return err
+	}
+	for _, cidr := range cidrs {
+		if _, _, err := net.ParseCIDR(cidr); err != nil {
```

Comment:
Can we check that this parses in convert.go or api/validation/validation.go?  It might be better to ensure things are in order before accepting the pod.

Reply:
Humm... I just found a need to validate an annotation as well and there wasn't scaffolding for doing that in api/validation/validate.go.  Here's the function I created:

```go
func ValidatePodAnnotations(annotations map[string]string, fldPath *field.Path) field.ErrorList {
	allErrs := field.ErrorList{}
	for k, v := range annotations {
		switch k {
		case apiannotations.PodResourcesPrivateIPOnly:
			_, err := strconv.ParseBool(v)
			fmt.Println("parsed bool")
			if err != nil {
				allErrs = append(allErrs, field.Invalid(fldPath.Child(k), v, "Could not parse annotation value as bool"))
			}
		}
	}
	return allErrs
}
```

Called in `ValidatePod`:

```go
allErrs = append(allErrs, ValidatePodAnnotations(pod.Annotations, field.NewPath("metadata.annotations"))...)
```

Reply:
:+1: